common/check_expired_locked_rules

```diff
@@ -1,56 +1,121 @@
+from sqlalchemy.sql import functions
+
+from rucio.db.sqla import models, session
+from utils.common import PrometheusPusher
 
 # Exit statuses
 OK, WARNING, CRITICAL, UNKNOWN = 0, 1, 2, 3
 
+if __name__ == '__main__':
 
-def main():
-    '''
-    Probe to check the locked expired rules or datasets with locked rules
-    '''
     status = OK
-    session = get_session()
-    try:
-        query = "select rawtohex(id), scope, name, rse_expression from atlas_rucio.rules where locked=1 and expires_at<sys_extract_utc(localtimestamp)"
-        print 'Locked expired rules'
-        for row in session.execute(query):
-            status = CRITICAL
-            print row[0], row[1], row[2]
-    except Exception as error:
-        print error
-        status = UNKNOWN
-        sys.exit(status)
-    try:
-        query = """select rawtohex(c.id), c.scope, c.name, c.rse_expression from atlas_rucio.rules c,
-                   (select a.scope, a.name from atlas_rucio.dids a
-                   where a.expired_at is not null and a.expired_at < sys_extract_utc(localtimestamp)
-                   and exists (select 1 from atlas_rucio.rules b where a.scope=b.scope and a.name=b.name and locked=1)) d
-                   where c.scope=d.scope and c.name=d.name and locked=1"""
-        print 'Datasets expired with locked rules'
-        for row in session.execute(query):
-            status = CRITICAL
-            print row[0], row[1], row[2], row[3]
-    except Exception as error:
-        print error
-        status = UNKNOWN
-        sys.exit(status)
-    sys.exit(status)
+    session = session.get_session()
+    with PrometheusPusher() as manager: 
+        try:            
+            statement = select(
+                models.ReplicationRule.id, 
+                models.ReplicationRule.scope, 
+                models.ReplicationRule.name, 
+                models.ReplicationRule.rse_expression
+            ).where(
+                and_(
+                    models.ReplicationRule.locked == '1', 
+                    models.ReplicationRule.expires_at<functions.current_timestamp()
+                )
+            )
+            query = session.execute(statement).scalars()
+            rule_counts = {"null":0}
+
+            print('Locked expired rules')
+            for rule_id, scope, name, rse_expression in query: 
+                print(rule_id, scope, name, rse_expression)
+                status = CRITICAL
+
+                if rse_expression in rule_counts.keys(): 
+                    rule_counts[rse_expression] += 1 
+                else: 
+                    rule_counts[rse_expression] = 1
+
+            for rse_expression, count in rule_counts.items():
+                manager.gauge('locked_expired_rules.{rse_expression}',
+                        documentation='Number of rules that are locked and expired, by RSE.'
+                    ).labels(
+                        rse_expression=rse_expression
+                    ).set(count)
```

Comment:
I’m not a huge fan of this styling. I believe PEP 8 would recommend:

```python
(manager.gauge('locked_expired_rules.{rse_expression}',
               documentation='Number of rules that are locked and expired, by RSE.')
        .labels(rse_expression=rse_expression)
        .set(count))
```

But, I can be persuaded.

Reply:
Do you have strong feelings on 
```
                (
                    manager.gauge(
                        "locked_expired_rules.{rse_expression}",
                        documentation="Number of rules that are locked and expired, by RSE.",
                    )
                    .labels(rse_expression=rse_expression)
                    .set(count)
                )
```
? Those double parenthesis make my skin itch .

Reply:
Please no, this is foul! :laughing:

Reply:
Then I'll counter-propose 
```
(manager.gauge(
        "locked_expired_rules.{rse_expression}",
        documentation="Number of rules that are locked and expired, by RSE expression.")
    .labels(rse_expression=rse_expression)
    .set(count))
```
Just because I really hate how 
```
obj.func(arg1, 
              arg2
)
```
looks, far more than I should.

Reply:
Yes, this is perfectly acceptable. I’ll take a moment to comment on the differences between the two styles for ‘academic purposes’.

The first difference is about how to place the parentheses and the arguments. Typically I follow two styles:
```python
foo = long_function_name(var_one, var_two,
                         var_three, var_four)

bar = long_function_name(
    var_one,
    var_two,
    var_three,
    var_four
)
```
However, PEP 8 endorses your style, so there is absolutely no argument against it.

The second difference is about indentation level. Generally, in a chained call of methods, I prefer to have them aligned:
```python
(instance.foo()
         .bar()
         .baz())
```
Your style breaks the alignment between the first two, but the amount of indentation does not depend on the length of the variable name, which is a plus.